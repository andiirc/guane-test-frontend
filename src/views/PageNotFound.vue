<template>
  <div class="message">
    <div class="message__img">
      <span>44</span>
    </div>
    <p>La página que está intentando buscar<br> no se encuentra en este universo.</p>
  </div>
</template>

<style lang="scss" >
.message{
  display: block;
  width: 100%;
  &__img{
    display: block;
    margin: 0 auto;
    transform: translateY(2%);
    width: fit-content;
    text-align: center;
    span {
      display: block;
      margin: 0 auto;
      color: $text-primary;
      font-weight: 600;
      font-size: 18em;
      display: block;
      overflow: hidden;
      width: fit-content;
      height: max-content;
      &:first-letter{
        letter-spacing: 21vmax;
      }
      &:before{
        content:'';
        background-image: url('~@/assets/images/ram.png');
        position: absolute;
        height: 100%;
        width: 100%;
        background-repeat: no-repeat;
        background-size: contain;
        background-position: center;
        animation: rotateIn 0.5s ease-out;
      }
    }
  }
  p{
    text-align: center;
    font-weight: 400;
    color: $text-tertiary;
  }
}

@keyframes rotateIn {
  from {
    transform: rotate( 0deg ) scale(0.2);
    opacity: 0;
  }
  to {
    transform: rotate( 360deg ) scale(1);
    opacity: 1;
  }
}

</style>
