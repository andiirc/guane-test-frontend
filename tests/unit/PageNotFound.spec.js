import { shallowMount } from '@vue/test-utils';
import PageNotFound from '@/views/PageNotFound.vue';

describe('PageNotFound', () => {
  it('render html', () => {
    const wrapper = shallowMount(PageNotFound);

    const render = wrapper.html();

    expect(render).toMatchSnapshot();
  });

  it('render exist class message', () => {
    const wrapper = shallowMount(PageNotFound);

    const render = wrapper.find('.message').exists();

    expect(render).toBe(true);
  });

  it('render text in paragraph', () => {
    const wrapper = shallowMount(PageNotFound);

    const render = wrapper.find('p').text();

    expect(render).toEqual('La página que está intentando buscar no se encuentra en este universo.');
  });

  it('render text in class message__img', () => {
    const wrapper = shallowMount(PageNotFound);

    const render = wrapper.find('.message__img').text();

    expect(render).toEqual('44');
  });

  it('contain text 44', () => {
    const wrapper = shallowMount(PageNotFound);

    const render = wrapper.html();

    expect(render).toContain('44');
  });

  it('render exist class message__img', () => {
    const wrapper = shallowMount(PageNotFound);

    const render = wrapper.find('.message__img').exists();

    expect(render).toBe(true);
  });
});
