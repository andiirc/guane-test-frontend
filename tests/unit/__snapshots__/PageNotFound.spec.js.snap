// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound render html 1`] = `
<div class="message">
  <div class="message__img"><span>44</span></div>
  <p>La página que está intentando buscar<br> no se encuentra en este universo.</p>
</div>
`;
